core/src/main/java/org/testcontainers/utility/LazyFuture.java

```diff
@@ -0,0 +1,44 @@
+package org.testcontainers.utility;
+
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.experimental.Delegate;
+import org.rnorth.ducttape.timeouts.Timeouts;
+
+import java.util.concurrent.*;
+
+/**
+ * Future implementation with lazy result evaluation <b>in the same Thread</b> as caller.
+ *
+ * @param <T>
+ */
+public abstract class LazyFuture<T> implements Future<T> {
+
+    @Delegate(excludes = Excludes.class)
+    private final Future<T> delegate = CompletableFuture.completedFuture(null);
+
+    @Getter(value = AccessLevel.MODULE, lazy = true)
```

Comment:
Ah nice - I noticed that the `lazyTransform`ed Future before didn't cache the result, so was going to suggest memoization.

Looks like this does the job concisely - another useful Lombok feature to learn about :)